### PR TITLE
fix(#24): fix failling test

### LINKED ISSUE #24: fix failling test on builds
Error: `Exception in thread "main" java.lang.UnsupportedClassVersionError: com/android/prefs/AndroidLocationsProvider has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
`

### PR DESCRIPTION
**Pull request by aloundoye**
fix #24

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -5,11 +5,11 @@ on: [push, pull_request]
 jobs:
   js-bootstrap:
     name: Test JS bootstrap on Node v${{ matrix.node-version }}
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-20.04
 
     strategy:
       matrix:
-        node-version: [ 12.x, 14.x, 16.x ]
+        node-version: [ 21.x, 22.x, 23.x ]
```
Comment (garethbowen) [inline .github/workflows/test.yml:12]: It's better to test even numbered node versions, because they're the ones that are supported for production. I suggest using 

```suggestion
        node-version: [ 18.x, 20.x, 22.x ]
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -26,11 +26,11 @@ jobs:
 
   js-library:
     name: Test JS library on Node v${{ matrix.node-version }}
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-20.04
 
     strategy:
       matrix:
-        node-version: [ 12.x, 14.x, 16.x ]
+        node-version: [ 21.x, 22.x, 23.x ]
```
Comment (garethbowen) [inline .github/workflows/test.yml:33]: As above.

File: gradlew [hunk 3]
```diff
@@ -81,92 +132,113 @@ Please set the JAVA_HOME variable in your environment to match the
...
+#
+
+eval "set -- $(
+        printf '%s\n' "$DEFAULT_JVM_OPTS $JAVA_OPTS $GRADLE_OPTS" |
+        xargs -n1 |
+        sed ' s~[^-[:alnum:]+,./:=@_]~\\&~g; ' |
+        tr '\n' ' '
+    )" '"$@"'
+
 exec "$JAVACMD" "$@"
```
Comment (garethbowen) [inline gradlew:244]: Why were these changes necessary?

Comment (aloundoye) [reply gradlew:244]: @garethbowen, these files were regenerated when I updated the Gradle version to `7.6`

**Review by garethbowen — APPROVED** [review]
Nice! Thanks
